pkg/snapstore/gcs_snapstore.go

```diff
@@ -60,16 +69,22 @@ const (
 // NewGCSSnapStore create new GCSSnapStore from shared configuration with specified bucket.
 func NewGCSSnapStore(config *brtypes.SnapstoreConfig) (*GCSSnapStore, error) {
 	ctx := context.TODO()
+	var emulatorConfig gcsEmulatorConfig
+	emulatorConfig.enabled = isEmulatorEnabled()
 	var opts []option.ClientOption // no need to explicitly set store credentials here since the Google SDK picks it up from the standard environment variable
 
-	if _, ok := os.LookupEnv(envSourceStoreCredentials); !ok { // do not set endpoint override when copying backups between buckets, since the buckets may reside on different regions
+	if _, ok := os.LookupEnv(envSourceStoreCredentials); !ok || emulatorConfig.enabled { // do not set endpoint override when copying backups between buckets, since the buckets may reside on different regions
 		endpoint := strings.TrimSpace(os.Getenv(envStorageAPIEndpoint))
 		if endpoint != "" {
 			opts = append(opts, option.WithEndpoint(endpoint))
+			emulatorConfig.endpoint = endpoint
 		}
 	}
-
-	if config.IsSource {
+	var chunkDirSuffix string
+	if emulatorConfig.enabled {
+		ConfigureGCSEmulator(&chunkDirSuffix, opts, emulatorConfig)
```

Comment:
I would suggest to not mutate chunkDirSuffix in this function. Instead, the function can simply be a method on the `emulatorConfig` struct, and can be called `configureClient`. This method should simply take the `opts` and add the new `option.WithoutAuthentication()` and also call `os.SetEnv()`. Additionally, the chunkSuffixDir should be done separately within the `if emulatorConfig.enabled{}` block, because this is etcdbr-snapstore specific configuration.